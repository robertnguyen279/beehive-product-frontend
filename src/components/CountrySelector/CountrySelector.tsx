import React from 'react';
import { Select } from 'antd';
import i18next from 'i18next';
import { useTranslation } from 'react-i18next';
import VietNamFlag from 'assets/icons/vietnam.svg';
import UnitedKingdomFlag from 'assets/icons/united-kingdom.svg';
import { useMediaQuery } from 'react-responsive';

const { Option } = Select;

const CountrySelector = () => {
  const { i18n } = useTranslation();
  const handleChange = (value: string): void => {
    i18n.changeLanguage(value);
  };
  const isMobile = useMediaQuery({ query: '(max-width: 575px)' });
  return (
    <Select
      defaultValue={i18next.language}
      style={
        isMobile
          ? { backgroundColor: 'transparent!important' }
          : { minWidth: '8rem', backgroundColor: 'transparent!important' }
      }
      onChange={handleChange}
    >
      <Option value="en">
        <div className="flex items-center justify-between relative">
          {!isMobile ? <div>English</div> : <div className="text-transparent w-5">.</div>}
          <img
            src={UnitedKingdomFlag}
            alt="english"
            className="w-5"
            style={
              isMobile
                ? { position: 'absolute', top: '50%', left: '50%', transform: 'translate(-50%, -50%)' }
                : undefined
            }
          />
        </div>
      </Option>
      <Option value="vi">
        <div className="flex items-center justify-between relative">
          {!isMobile ? <div>Tiếng Việt</div> : <div className="text-transparent w-5">.</div>}
          <img
            src={VietNamFlag}
            alt="vietnamese"
            className="w-5 "
            style={
              isMobile
                ? { position: 'absolute', top: '50%', left: '50%', transform: 'translate(-50%, -50%)' }
                : undefined
            }
          />
        </div>
      </Option>
    </Select>
  );
};

export default CountrySelector;
