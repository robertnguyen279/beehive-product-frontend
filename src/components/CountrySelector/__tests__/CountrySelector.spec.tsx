import React from 'react';
import { shallow } from 'enzyme';

import CountrySelector from '../CountrySelector';
// const clickFn = jest.fn();

describe('CountrySelector', () => {
  const component = shallow(<CountrySelector />);

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
