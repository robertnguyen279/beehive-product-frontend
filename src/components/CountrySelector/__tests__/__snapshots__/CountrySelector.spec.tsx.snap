// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountrySelector should render correctly 1`] = `
<ForwardRef(InternalSelect)
  onChange={[Function]}
  style={
    Object {
      "backgroundColor": "transparent!important",
      "minWidth": "8rem",
    }
  }
>
  <Option
    value="en"
  >
    <div
      className="flex items-center justify-between relative"
    >
      <div>
        English
      </div>
      <img
        alt="english"
        className="w-5"
        src="test-file-stub"
      />
    </div>
  </Option>
  <Option
    value="vi"
  >
    <div
      className="flex items-center justify-between relative"
    >
      <div>
        Tiếng Việt
      </div>
      <img
        alt="vietnamese"
        className="w-5 "
        src="test-file-stub"
      />
    </div>
  </Option>
</ForwardRef(InternalSelect)>
`;
